tensorflow_addons/seq2seq/basic_decoder.py

```diff
@@ -43,7 +43,54 @@ class BasicDecoderOutput(
 
 
 class BasicDecoder(decoder.BaseDecoder):
-    """Basic sampling decoder."""
+    """Basic sampling decoder for training and inference.
+
+    The `tfa.seq2seq.Sampler` instance passed as argument is responsible to sample from
+    the output distribution and produce the input for the next decoding step. The decoding
+    loop is implemented by the decoder in its `__call__` method.
+
+    Example using `tfa.seq2seq.TrainingSampler` for training:
+
+    >>> batch_size = 4
+    >>> max_time = 7
+    >>> hidden_size = 32
+    >>> vocab_size = 64
+    >>>
+    >>> embedding_layer = tf.keras.layers.Embedding(vocab_size, hidden_size)
```

Comment:
We might want to choose a different value for Embedding layer since the input vocab and state size could be different from the output decode/vocab size. The current example might give user a impression that they need to be same.